Add Nest Camera Extension for Raycast

chadrwalters opened the pull request:
## Description

This PR adds a new Nest Camera extension for Raycast that allows users to view and manage their Nest cameras directly from Raycast.

### Features
- View all Nest cameras in your account
- Stream camera feeds using RTSP protocol
- Quick access to camera streams from the menu bar
- Support for Google Nest API authentication
- Automatic token refresh and session management

### Technical Details
- Uses Google's Device Access API for Nest devices
- Implements RTSP streaming with FFmpeg for reliable video playback
- Handles OAuth2 authentication flow with Google
- Provides proper error handling and user feedback
- Includes comprehensive type definitions for API responses

### Screenshots
![nest-raycast-extension-1](https://github.com/user-attachments/assets/274be81a-98fd-4893-a880-1e78b6e53a39)
![nest-raycast-extension-2](https://github.com/user-attachments/assets/8e59ac5d-d099-4473-83e1-5691c738b8d4)
![nest-raycast-extension-3](https://github.com/user-attachments/assets/fcd49a44-0f67-4d45-963f-117089383faf)
![nest-raycast-extension-4](https://github.com/user-attachments/assets/3571f538-f4c8-4719-9da2-5b5e87bdcdf8)
![nest-raycast-extension-5](https://github.com/user-attachments/assets/fee5a009-4cc6-4f90-a609-0978170491a5)


## Checklist
- [x] Extension follows Raycast design guidelines
- [x] All linting issues have been fixed
- [x] Type safety has been improved throughout the codebase
- [x] Documentation has been added
- [x] Extension has been tested on macOS

pernielsentikaer reviewed (commented):
We can't just bundle a binary in the extension like that, see more here: https://developers.raycast.com/basics/prepare-an-extension-for-store#binary-dependencies-and-additional-configuration

chadrwalters commented:
I've completed the restructuring of the Nest Camera extension to comply with Raycast's guidelines. Here's a summary of the changes:

### Removed Bundled Binary Files
- **Definition**: Elimination of pre-packaged binary applications from the extension
- **How to Track**: Verify absence of Platypus app binary and related build scripts
- **Why It Matters**: Complies with Raycast's guidelines against distributing third-party binaries
- **Special Consideration**: Ensures extension only uses system-installed dependencies

### Updated Implementation to Use FFplay Directly
- **Definition**: Refactored code to utilize the FFplay utility installed on user's system
- **How to Track**: Review nest-player.sh script for direct FFplay calls with optimized parameters
- **Why It Matters**: Creates a more transparent and maintainable implementation
- **Special Consideration**: Includes robust FFplay detection across multiple common installation paths

### Enhanced Error Handling and User Experience
- **Definition**: Improved feedback mechanisms and process management
- **How to Track**: Monitor error messages when FFplay is missing and observe loading dialogs
- **Why It Matters**: Provides clear guidance to users when requirements aren't met
- **Special Consideration**: Includes better video window positioning and process cleanup

### Updated Documentation
- **Definition**: Revised README and supporting documentation
- **How to Track**: Check for clear FFmpeg installation instructions and accurate limitations
- **Why It Matters**: Sets proper expectations and provides necessary setup guidance
- **Special Consideration**: Removes references to previously bundled applications

### Additional Improvements
- **Definition**: Supplementary enhancements to overall extension quality
- **How to Track**: Review environment variable handling and logging implementation
- **Why It Matters**: Ensures reliable operation within Raycast environment
- **Special Consideration**: Code now adheres to Raycast's style guidelines

The extension now works correctly while complying with Raycast's guidelines. The implementation detects if FFplay is installed and provides helpful error messages if it's missing.

chadrwalters commented:
> We can't just bundle a binary in the extension like that, see more here: https://developers.raycast.com/basics/prepare-an-extension-for-store#binary-dependencies-and-additional-configuration

Removed

pernielsentikaer commented:
Can you also record a screencast of this?

chadrwalters commented:
I've addressed all the review comments and made several improvements to the codebase. Here's a summary of the changes:

### High Priority Fixes
1. **Bundled Binary Removal**
   - Removed the Platypus app binary
   - Updated implementation to use FFplay directly
   - Added robust FFplay detection across common installation paths
   - Improved error messages for missing FFplay

2. **Device ID Format Standardization**
   - Added `getFullDeviceId` helper method in `WebRTCStreamService`
   - Standardized device ID format across storage and API calls
   - Ensures consistent identifier usage throughout the codebase

3. **Rate Limit Error Handling**
   - Added specific handling for 429 (rate limit) errors in `extendStream`
   - Improved error logging with detailed context
   - Consistent error handling pattern across methods

### Medium Priority Fixes
1. **Code Quality Improvements**
   - Removed unused `StreamOptions` interface
   - Cleaned up redundant validation in `getConfig`
   - Standardized error handling patterns
   - Improved type safety throughout

2. **Dependency Management**
   - Pinned hls.js to version 1.5.1 for stability
   - Removed unnecessary dependencies
   - Updated documentation to reflect changes

### Documentation Updates
1. **README and Configuration**
   - Updated repository URL
   - Added clear FFmpeg installation instructions
   - Improved error message clarity
   - Updated configuration documentation

### Testing and Verification
- All linting issues have been resolved
- Code follows Raycast design guidelines
- Type safety has been improved throughout
- Extension tested on macOS
- All PR review comments have been addressed

The extension now provides a more robust and maintainable implementation while complying with Raycast's guidelines. Would you like me to explain any of these changes in more detail?

pernielsentikaer commented on extensions/nestcamera/README.md:
```diff
@@ -0,0 +1,217 @@
+   - Complete registration ($5 one-time fee)
+   - Create a project and note the Project ID
+
+4. Configure OAuth2:
```
I tried to set this up but I keep getting an error

![Raycast 2025-03-06 at 09 53 53](https://github.com/user-attachments/assets/5da2ad91-fb4b-4777-96c8-bea4a87610aa)

I tried to follow all steps, something like OAuth client ID application type could be nice
